File: src/skillguard/reporting/json_report.py
"""JSON report generator."""

from __future__ import annotations

import json

from skillguard.core.models import ScanResult


def generate_json_report(result: ScanResult) -> str:
    """Generate a JSON report from scan results.

    Returns:
        Pretty-printed JSON string.
    """
    return result.model_dump_json(indent=2)


def generate_json_summary(result: ScanResult) -> str:
    """Generate a concise JSON summary (no findings detail)."""
    data = result.model_dump()
    # Remove detailed findings for summary
    for er in data.get("engine_results", []):
        er["findings"] = []
        er["findings_count"] = len(result.engine_results)
    return json.dumps(data, indent=2, default=str)

File: go-skillguard/internal/reporting/json_report.go
// Package reporting provides report generation in various formats.
package reporting

import (
	"encoding/json"

	"github.com/girdav01/skillguard/internal/core"
)

// GenerateJSONReport generates a JSON report from scan results.
func GenerateJSONReport(result *core.ScanResult) (string, error) {
	data, err := json.MarshalIndent(result, "", "  ")
	if err != nil {
		return "", err
	}
	return string(data), nil
}

// GenerateJSONSummary generates a concise JSON summary without detailed findings.
func GenerateJSONSummary(result *core.ScanResult) (string, error) {
	type summaryEngine struct {
		EngineName    string             `json:"engine_name"`
		EngineVersion string             `json:"engine_version"`
		Verdict       core.EngineVerdict `json:"verdict"`
		Confidence    float64            `json:"confidence"`
		FindingsCount int                `json:"findings_count"`
		DurationMs    int64              `json:"duration_ms"`
	}

	engines := make([]summaryEngine, len(result.EngineResults))
	for i, er := range result.EngineResults {
		engines[i] = summaryEngine{
			EngineName:    er.EngineName,
			EngineVersion: er.EngineVersion,
			Verdict:       er.Verdict,
			Confidence:    er.Confidence,
			FindingsCount: len(er.Findings),
			DurationMs:    er.DurationMs,
		}
	}

	summary := map[string]any{
		"scan_id":              result.ScanID,
		"skill_name":           result.SkillName,
		"composite_score":      result.CompositeScore,
		"verdict":              result.Verdict,
		"total_findings":       result.TotalFindings,
		"findings_by_severity": result.FindingsBySeverity,
		"files_scanned":        result.FilesScanned,
		"engine_results":       engines,
	}

	data, err := json.MarshalIndent(summary, "", "  ")
	if err != nil {
		return "", err
	}
	return string(data), nil
}
